functions.cpp

```diff
@@ -60,12 +60,15 @@ int str_to_int(std::string str) {
 }
 
 std::vector<std::string> split_string(const std::string& str, const std::string& separator) {
-    std::vector<std::string> SplitVec;
+    std::vector<std::string> SplitVec,  to_return;
```

Comment:
Why do we need a new allocation? Can't we trim in-place?

Reply:
Done